app/dlq/infrastructure/mq/listeners/dlq_queue_listener_base.py

```
@@ -0,0 +1,67 @@
+import os
+from abc import ABC
+
+from app.dlq.infrastructure.mq.exceptions.mq_exception import MqException
+from app.dlq.infrastructure.mq.listeners.stomp_listener_base import StompListenerBase
+from app.dlq.infrastructure.mq.publishers.resubmitting_publisher_base import ResubmittingPublisherBase
+
+
+class DlqQueueListenerBase(StompListenerBase, ABC):
+
+    def __init__(self, resubmitting_publisher: ResubmittingPublisherBase) -> None:
+        super().__init__()
+        self.__resubmitting_publisher = resubmitting_publisher
+
+    def _handle_received_message(self, message_body: dict) -> None:
+        self._logger.info("Received message from DLQ Queue. Message body: {}".format(str(message_body)))
+
+        message_admin_metadata = message_body.get('admin_metadata')
+        if message_admin_metadata is None:
+            self._logger.info("Received message does not include admin_metadata. Ignoring message...")
+            return
+
+        original_queue = message_admin_metadata.get('original_queue')
+        if original_queue is None:
+            self._logger.info(
+                "Received message does not include original_queue inside admin_metadata. Ignoring message..."
+            )
+            return
+
```

Should invalid messages, the ones without admin_metadata or the original_queue be discarded automatically?

Wouldn't this hide errors as messages will disappear from the DLQ?

Should these messages be examined manually and discarded by an ops person?

---

@GPortas this is a good point.  If you use client-individual for acknowledgement, you can manually send the ack once everything has been validated.  If it fails, it can be left and an email sent to the admins.

---

Thank you very much for your very valuable suggestions. I have implemented it that way.

Emailing to administrators will be implemented in HDC-217.